### ISSUE TITLE
Move crates in `gloo` behind features

### ISSUE THREAD

**Issue by hamza1311** [rustwasm/gloo#137]
Move the crates in gloo behind feature locks. This means
- `crates/*` will go away
- There will be modules for each feature
- Modules will be behind feature locks

**Comment by DesmondWillowbrook** [comment]
The purpose of feature gating is to prevent unwanted dependencies from sneaking in, but we already have that represented as individual crates. (i.e. `gloo_file` instead of `gloo::file`) The `gloo` crate, therefore, exists primarily for convenience of being able to pull in whatever modules one wants, _without_ reading about all the features gates, etc.

**Comment by hamza1311** [comment]
I meant that we should remove the `gloo-*` crates and move their code under feature gated modules in gloo

**Comment by mkroehnert** [comment]
Hi, I have not used the gloo crates extensively but I have seen them used in other projects that often use just one of the gloo-* crates instead of the complete gloo package.

For these projects, moving all gloo-* libs into a single gloo crate would mean
* no notification and update via e.g. `cargo update` once the combined crate is published
* you will have to update all gloo libraries at once, if a new version comes out, which you might not want to do, if for example you want to add a new lib but do not want to update another existing one due to  breaking/incompatible changes
* is it a valid usecase, to want a callback-based API from one and a futures-based API from two different gloo-* crates in one project? If so, how could this be achieved via features and single crate?

Not 100% sure though, whether the second point occurs often or even is a valid usecase.

**Comment by lukechu10** [comment]
I agree with @mkroehnert on this one. In addition, this will likely increase compilation time because cargo can no longer compile them in parallel.

**Comment by hamza1311** [comment]
I agree with those points too. I originally mentioned to allow having `gloo::utils` module for #72. It seems like it's a better idea to crate `gloo-utils` instead of that.
